src/lib.rs

```diff
@@ -93,9 +94,12 @@ fn get_lines(
     current_dir: &PathBuf,
 ) -> Result<BTreeMap<FileEntry, Vec<String>>, Box<dyn Error>> {
     let file_paths: Vec<PathBuf> = get_needed_file_paths(args);
+    // underscore, followed by 10 or more decimals (timestamp) and followed by .bak
+    let backup_file_regex = Regex::new(r"_\d{10,}\.bak$")?;
```

Comment:
I think you can do it easier. Just extend the method [is_env_file](https://github.com/dotenv-linter/dotenv-linter/blob/master/src/common/file_entry.rs#L47):
```rust
pub fn is_env_file(path: &PathBuf) -> bool {
    let pattern = ".env";
    Self::get_file_name(path)
        .filter(|file_name| !EXCLUDED_FILES.contains(file_name))
        .filter(|file_name| file_name.starts_with(pattern) || file_name.ends_with(pattern))
        .filter(|file_name| !file_name.ends_with(".bak")) # NEW LINE
        .is_some()
}
```